src/compiler/filesystem_local_win.cc

```diff
@@ -62,19 +62,33 @@ static int root_prefix_length(const char* path) {
   }
   int length = strlen(path);
   if (length == 0) return 0;
-  if (is_path_separator(path[0])) {
+  if (is_path_separator_(path[0])) {
     if (length == 1) return 0;  // Drive root is not absolute.
     if (path[1] == path[0]) return 2;  // Network path.
       return 0;
   }
   if (length < 3) return 0;
   bool is_ascii_drive = ('a' <= path[0] && path[0] <= 'z') || ('A' <= path[0] && path[0] <= 'Z');
-  if (is_ascii_drive && path[1] == ':' && is_path_separator(path[2])) {
+  if (is_ascii_drive && path[1] == ':' && is_path_separator_(path[2])) {
     return 3;  // Drive root.
   }
   return 0;
 }
 
+const char* FilesystemLocal::relative_anchor(const char* path) {
+  ASSERT(!is_absolute(path));
+  if (path[0] == '/') {  // Safe to access even for empty strings.
+    // TODO(florian): error checking.
+    DWORD result_length = GetFullPathName("/", 0, NULL, NULL);
```

Comment:
Can this be cached in any meaningful way? Looks like a ton of allocations.

Reply:
I would like to revisit this area again.
The current setup (mostly?) fixes the immediate issues, but I hope to find a better API.